### ISSUE TITLE
libsecp256k1 dependency

### ISSUE THREAD

**Issue by PabloCastellano** [libbitcoin/libbitcoin-system#31]
Long time since pulling and I have just found that I don't have this dependency (libsecp256k1) and have no idea about how to install it or where to get it.

Please document it :)

  CXX      utility/ec_keys.lo
utility/ec_keys.cpp:20:23: fatal error: secp256k1.h: No such file or directory
 #include <secp256k1.h>
                       ^
compilation terminated.

**Comment by erm3nda** [comment]
I've installed that as suggested, and now img getting:
`error: ‘secp256k1_context_t’ does not name a type.`

Suggestion?

**Comment by evoskuil** [comment]
Detailed installation instructions are provided here:
https://github.com/libbitcoin/libbitcoin/blob/version2/install.sh

See specifically:
https://github.com/libbitcoin/libbitcoin/blob/version2/install.sh#L570

The secp256k1 library is currently a moving target, the interface changes without version changes. This is why we maintain and build from a fork.

**Comment by thracia91** [comment]
I am trying to build libbitcoin on Fedora 25 (I downloaded the install script from the "version2" fork) and receive this error:

checking for secp256k1... no
configure: error: Package requirements (libsecp256k1 >= 0.0.1) were not met:
No package 'libsecp256k1' found

These OpenSSL packages are installed:

openssl.x86_64                         1:1.0.2j-3.fc25                  @updates
openssl-devel.x86_64                   1:1.0.2j-3.fc25                  @updates
openssl-libs.i686                      1:1.0.2j-3.fc25                  @updates
openssl-libs.x86_64                    1:1.0.2j-3.fc25                  @updates

and this is the output from "openssl ecparam -list_curves":

 secp256k1 : SECG curve over a 256 bit prime field
  secp384r1 : NIST/SECG curve over a 384 bit prime field
  secp521r1 : NIST/SECG curve over a 521 bit prime field
  prime256v1: X9.62/SECG curve over a 256 bit prime field

In other words, secp256k1 is indeed installed and available on the system but the build script can't seem to find the library. Anyone have any idea how to solve this?

**Comment by RojavaCrypto** [comment]
Does `pkg-config --cflags --libs libsecp256k1` show anything?

**Comment by evoskuil** [comment]
Libbitcoin does not use OpenSSL.

The install script (install.sh) always downloads and builds libsecp256k1 so it is hard to see how this could happen without preceding failures.

**Comment by thracia91** [comment]
This is the output from `pkg-config --cflags --libs libsecp256k1` : 

```
Package libsecp256k1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libsecp256k1.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libsecp256k1' found
```

There are no messages ahead of "checking for secp..." that indicate any sort of failure or warning.

**Comment by evoskuil** [comment]
This indicates that the correct library was built and installed into `/usr/local/lib/` by the installer:
```
make[1]: Entering directory '/home/eric/build-libbitcoin/secp256k1'
 /usr/bin/mkdir -p '/usr/local/lib'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libsecp256k1.la '/usr/local/lib'
libtool: install: /usr/bin/install -c .libs/libsecp256k1.so.0.0.0 /usr/local/lib/libsecp256k1.so.0.0.0
libtool: install: (cd /usr/local/lib && { ln -s -f libsecp256k1.so.0.0.0 libsecp256k1.so.0 || { rm -f libsecp256k1.so.0 && ln -s libsecp256k1.so.0.0.0 libsecp256k1.so.0; }; })
libtool: install: (cd /usr/local/lib && { ln -s -f libsecp256k1.so.0.0.0 libsecp256k1.so || { rm -f libsecp256k1.so && ln -s libsecp256k1.so.0.0.0 libsecp256k1.so; }; })
libtool: install: /usr/bin/install -c .libs/libsecp256k1.lai /usr/local/lib/libsecp256k1.la
libtool: install: /usr/bin/install -c .libs/libsecp256k1.a /usr/local/lib/libsecp256k1.a
libtool: install: chmod 644 /usr/local/lib/libsecp256k1.a
libtool: install: /usr/bin/ranlib /usr/local/lib/libsecp256k1.a
```
This indicates that headers were published and it was registered with package config in the same directory:
```
 /usr/bin/mkdir -p '/usr/local/include'
 /usr/bin/install -c -m 644 include/secp256k1.h include/secp256k1_recovery.h '/usr/local/include'
 /usr/bin/mkdir -p '/usr/local/lib/pkgconfig'
 /usr/bin/install -c -m 644 libsecp256k1.pc '/usr/local/lib/pkgconfig'
```
This indicates that the system package config path is not set to include `/usr/local/lib/`:
```
Perhaps you should add the directory containing `libsecp256k1.pc'
to the PKG_CONFIG_PATH environment variable
```
That is not typical. You can either follow the advice in the message above or use the `--prefix` installer option.

**Comment by thracia91** [comment]
Thanks! Build was successful using --prefix=/usr/local/bin/.
